src/main/java/org/cloudfoundry/promregator/rewrite/AbstractMetricFamilySamplesEnricher.java

```diff
@@ -52,6 +51,6 @@ public AbstractMetricFamilySamplesEnricher() {
 	
 	public abstract List<String> getEnrichedLabelValues(List<String> original);
 
-	protected abstract void removeDuplicateLabels(List<String> original);
-	
+	protected abstract List<String> getEnrichedLabelValues(List<String> originalLabelNames, List<String> originalLabelValues);
```

Comment:
Do you have any special reason why you doubled the method? Looks rather complicated to me this way...

Reply:
I was worried about doing unnecessary work for anywhere else this method is called and not used, but now that i look again it appears to only be the tests that call this method.  I will remove the overload

Reply:
Actually i was wrong.  There are other classes (getOwnTelemetry, etc) that call it as well.  The iteration, and contains (using the HashSet) shouldn't have an impact of the performance.